datafusion/src/optimizer/remove_duplicate_filters.rs

```diff
@@ -0,0 +1,310 @@
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! Remove duplicate filters optimizer rule
+
+use crate::execution::context::ExecutionProps;
+use crate::logical_plan::Expr;
+use crate::logical_plan::LogicalPlan;
+use crate::optimizer::optimizer::OptimizerRule;
+use crate::optimizer::utils;
+use crate::optimizer::utils::optimize_explain;
+use crate::{error::Result, logical_plan::Operator};
+
+/// Remove duplicate filters optimizer.
+/// # Introduction
+/// It uses boolean algebra laws to simplify or reduce the number of terms in expressions.
+///
+/// Filter: #b Gt Int32(2) And #b Gt Int32(2)
+/// is optimized to
+/// Filter: #b Gt Int32(2)
+pub struct RemoveDuplicateFilters {}
+
+fn expr_contains<'a>(expr: &'a Expr, needle: &'a Expr) -> bool {
+    match expr {
+        Expr::BinaryExpr {
+            left,
+            op: Operator::And,
+            right,
+        } => expr_contains(left, needle) || expr_contains(right, needle),
+        Expr::BinaryExpr {
+            left,
+            op: Operator::Or,
+            right,
+        } => expr_contains(left, needle) || expr_contains(right, needle),
+        _ => expr == needle,
+    }
+}
+
+fn as_binary_expr<'a>(expr: &'a Expr) -> Option<&'a Expr> {
+    match expr {
+        Expr::BinaryExpr { .. } => Some(expr),
+        _ => None,
+    }
+}
+
+fn simplify<'a>(expr: &'a Expr) -> Expr {
+    match expr {
+        Expr::BinaryExpr { left, op: _, right } if left == right => simplify(left),
```

Comment:
This seems not correct to me, it seems to "simplify" `a + a` to `a` too?

Reply:
I am seeing one other example from the test `1=1` is also a good example, it now simplifies to `1`

Reply:
You're right, I was wrongly assuming that all expressions are boolean so ... my mistake.
I have updated. I hope I can check your other comments soon. Thanks !